lib/views/topLevelView.js

```diff
@@ -0,0 +1,351 @@
+
+var loadTemplate = require('../loadTemplate')
+
+var summarizeSBOL = require('./utils/summarizeSBOL')
+
+var async = require('async')
+
+var pug = require('pug')
+
+var sparql = require('../sparql/sparql-collate')
+
+var config = require('../config')
+
+var URI = require('sboljs').URI
+
+var getUrisFromReq = require('../getUrisFromReq')
+
+const uriToUrl = require('../uriToUrl')
+
+const request = require('request')
+
+var postprocess_igem = require('../postprocess_igem')
+
+var striptags = require('striptags')
+
+var generateDataRecord = require('../bioschemas/DataRecord')
+
+var extend = require('xtend')
+
+module.exports = function(req, res, type) {
+
+    typeShortName = 'Unknown'
+    if (type.lastIndexOf('#')>=0 && type.lastIndexOf('#')+1 < type.length) {
+	typeShortName = type.slice(type.lastIndexOf('#')+1)
+	if (typeShortName === 'Component') typeShortName = 'ComponentInstance'
+	else if (typeShortName === 'Module') typeShortName = 'ModuleInstance'
+	else if (typeShortName === 'ComponentDefinition') typeShortName = 'Component'
+	else if (typeShortName === 'ModuleDefinition') typeShortName = 'Module'
+    } else if (type.lastIndexOf('/')>=0 && type.lastIndexOf('/')+1 < type.length) {
+	typeShortName = type.slice(type.lastIndexOf('/')+1)
+    }
+    
+    var locals = {
+        config: config.get(),
+        section: typeShortName,
+        user: req.user
+    }
+
+    var meta
+    var sbol
+    var topLevel
+    var collectionIcon 
+    var remote
+
+    var collections = []
+    var submissionCitations = []
+    var citations = []
+    var builds = []
+
+    var encodedProteins = []
+    var otherComponents = []
+    var mappings = {}
+
+    const { graphUri, uri, designId, share, url } = getUrisFromReq(req, res)
+
+    var templateParams = {
+        uri: uri
+    }
+
+    var getCitationsQuery = loadTemplate('sparql/GetCitations.sparql', templateParams)
+
+    var getBuildsQuery = loadTemplate('sparql/GetImplementations.sparql', templateParams)
+
+    fetchSBOLObjectRecursive(uri, graphUri).then((result) => {
+	
+        sbol = result.sbol
+        topLevel = result.object
+        remote = result.remote
+
+        if(!topLevel || topLevel instanceof URI) {
+            locals = {
+                config: config.get(),
+                section: 'errors',
+                user: req.user,
+                errors: [ uri + ' Record Not Found: ' + topLevel ]
+            }
+            res.send(pug.renderFile('templates/views/errors/errors.jade', locals))
+            return Promise.reject()
+        }
+	meta = summarizeSBOL(typeShortName,topLevel,req,sbol,remote,graphUri)
+	
+    }).then(function lookupCollections() {
+
+        return Promise.all([
+	    getContainingCollections(uri, graphUri, req.url).then((_collections) => {
+
+		collections = _collections
+
+		collections.forEach((collection) => {
+
+		    collection.url = uriToUrl(collection.uri)
+
+                    const collectionIcons = config.get('collectionIcons')
```

Comment:
Could move outside this function so that we do not read from disk/context switch for every collection